kbfscrypto/encrypted_data.go

```diff
@@ -228,3 +228,59 @@ func DecryptTLFCryptKeys(
 
 	return oldKeys, nil
 }
+
+// EncryptedMerkleLeaf is an encrypted MerkleLeaf object.
+type EncryptedMerkleLeaf struct {
+	encryptedData
+}
+
+// MakeEncryptedMerkleLeaf constructs an EncryptedMerkleLeaf.
+func MakeEncryptedMerkleLeaf(
+	version EncryptionVer, data []byte, nonce *[24]byte) EncryptedMerkleLeaf {
+	return EncryptedMerkleLeaf{
+		encryptedData{
+			Version:       version,
+			EncryptedData: data,
+			Nonce:         nonce[:],
+		},
+	}
+}
+
+// PrepareMerkleLeaf verifies the correctness of the given leaf, and
+// returns its nonce.
+func PrepareMerkleLeaf(encryptedMerkleLeaf EncryptedMerkleLeaf) (
+	nonce [24]byte, err error) {
+	if encryptedMerkleLeaf.Version != EncryptionSecretbox {
+		return [24]byte{},
```

Comment:
Not a big deal since the optimizer will probably take care of it, but this could just be `nonce`.